### DISCUSSION TITLE
Affine registration with VoxelMorph

### DISCUSSION THREAD

**Discussion by LauraConnolly** [voxelmorph/voxelmorph#253] [Q&A]
Hi! I'm trying to use VoxelMorph on my own dataset and it's working well for deformable registration but I noticed the tutorial said "Rigid alignment is also possible with VoxelMorph, but it's not our focus here. A separate tutorial is being designed for that." and I wanted to try this as well. I can't seem to find the separate tutorial so I was wondering if someone could point me in the right direction or knows what to modify from that tutorial to perform rigid alignment instead?

Thanks in advance!!

**Comment by konstantlem** [comment]
Hello, 

I am also interested in linear registration. Is there any update on this?

Thank you!

**Comment by hkabbech** [comment]
Hi,

I am also looking for the tutorial for "rigid" alignment. Would be nice to get some information?

Thanks

**Comment by dalbis** [comment]
Similar question here: can Voxelmorph do affine registration? If yes, are there instructions for that? Thanks

**Reply by ktdiedrich** [reply]
I'm also looking for affine or rigid registration in voxelmorph. I found an affine transform class https://github.com/voxelmorph/voxelmorph/blob/dev/voxelmorph/tf/networks.py#L997
And it gets used in scripts and tests a few times, these might be examples
https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/register.py#L67
https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/test.py#L84
https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/train_instance.py#L113
https://github.com/voxelmorph/voxelmorph/blob/dev/scripts/tf/warp.py#L58

**Comment by mu40** [comment]
We just released a SynthMorph tutorial on anatomy-aware affine registration at [w3id.org/synthmorph](https://w3id.org/synthmorph), eliminating the need for segmentation prior to registration (such as skull-stripping). The network used in the tutorial has a flag for switching from affine to rigid registration, too.
